libs/components/datetime/src/lib/modules/date-range-picker/types/date-range-calculator.ts

```diff
@@ -10,36 +13,53 @@ import { SkyDateRangeCalculatorType } from './date-range-calculator-type';
  * Represents the calculator.
  */
 export class SkyDateRangeCalculator {
+  #config: SkyDateRangeCalculatorConfig;
+  #resourcesSvc: SkyLibResourcesService;
+
   /**
    * The text to display in the calculator select menu.
    */
-  public shortDescription: string;
+  public get shortDescription$(): Observable<string> {
+    if (this.shortDescription) {
+      return of(this.shortDescription);
+    }
+
+    if (!this.#shortDescriptionResourceKey) {
+      throw new Error(
+        'Calculator created without short description or resource key.',
+      );
+    }
```

Comment:
This effectively makes `shortDescription` or `shortDescription$` required right? I'm all for that change but I'm trying to think of how to communicate that to consumers.

it wouldnt make sense for consumers to create a calculator without a short description, but today the input property `shortDescription` is marked as optional. Do we have something in the docs to mark that one of two possible options needs to be provided?

Reply:
The `shortDescription` is already required in the `SkyDateRangeCalculatorConfig` interface, so all custom calculators should be initialized with a short description already. The property we're deprecating is the `shortDescription` property found on the _generated_ calculator; we're essentially deprecating the method used to access the `shortDescription` value. Since the short description is an async value, we should move consumers away from using the static `shortDescription` property and recommend they use the new observable instead. However, keep in mind that accessing the short description of a custom calculator is a niche feature and should rarely be used. That value is primarily used by the date range calculator select menu.

Reply:
The error is more or less a "safety check" for the type checker since consumers should never get in that scenario (and should never provide a resources key, since that's an internal detail).